### PR TITLE
Reconcile PDBs on generation field

### PR DESCRIPTION
**Pull request by iholder-redhat**
**What this PR does / why we need it**:
Reconcile PodDisruptionBudgets that are generated based on deployments on generation field.
    
The PDBs that should be effected are:
- virt-controller-pdb
- virt-api-pdb

The expected generation is set on the KV cr. If the generation of the cached PDB is different than the expected or if the metadata in the cached PDB does not match the expected the resource is updated.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by iholder-redhat** [comment]
Dependent upon #4746.
/hold

**Comment by iholder-redhat** [comment]
> Dependent upon #4746.
> /hold

Merged.
/hold cancel

File: pkg/virt-operator/resource/apply/generations.go [hunk 1]
```diff
@@ -1,52 +1,77 @@
...
+	switch required.(type) {
+	case *admissionregistrationv1beta1.MutatingWebhookConfiguration:
+		group = "admissionregistration.k8s.io"
+		resource = "mutatingwebhookconfigurations"
+	case *admissionregistrationv1beta1.ValidatingWebhookConfiguration:
+		group = "admissionregistration.k8s.io"
+		resource = "validatingwebhookconfigurations"
+	case *v1beta1.PodDisruptionBudget:
+		group = "apps"
+		resource = "poddisruptionbudget"
```
Comment (ashleyschuett) [inline pkg/virt-operator/resource/apply/generations.go:33]: `poddisruptionbudget` -> `poddisruptionbudgets`

**Comment by iholder-redhat** [comment]
Unit tests pass locally
/test pull-kubevirt-unit-test

**Review by ashleyschuett — COMMENTED** [review]
looks good! Once last thing, lets add 2 unit tests to to make sure if a PDB is changed it gets updated and if it's not that it doesn't.

**Comment by kbidarkar** [comment]
> looks good! Once last thing, lets add 2 unit tests to to make sure if a PDB is changed it gets updated and if it's not that it doesn't.

Was thinking we should have a test-case For PDB under tests/operator_test.go
Probably should go in here, https://github.com/kubevirt/kubevirt/blob/master/tests/operator_test.go#L889

So that we can run these tests in Downstream.

**Background:** I have a dropped a mail about it.

**Comment by iholder-redhat** [comment]
> > looks good! Once last thing, lets add 2 unit tests to to make sure if a PDB is changed it gets updated and if it's not that it doesn't.
> 
> Was thinking we should have a test-case For PDB under tests/operator_test.go
> Probably should go in here, https://github.com/kubevirt/kubevirt/blob/master/tests/operator_test.go#L889
> 
> So that we can run these tests in Downstream.
> 
> **For more Details:** I have a dropped a mail about it.

A functional test for PDB is now added in `operator_test.go` (in [this commit](https://github.com/kubevirt/kubevirt/pull/5144/commits/bc87a2050f15186494196d5882918e52e5deb335)).

**Comment by iholder-redhat** [comment]
Test constantly passes locally..
/retest

**Comment by iholder-redhat** [comment]
tests pass locally
/test pull-kubevirt-unit-test

**Comment by iholder-redhat** [comment]
Rebased.

**Pull request merged**